BEP12.md

```diff
@@ -72,6 +72,59 @@ return nil
 }
 ```
 
+### 0x2: Whitelisted Sender
+This script is used to ensure transfer transactions have valid recipients.
+The script will reject transfers to addresses that are not currently whitelisted.
+
+Firstly, this script will check the following conditions:
+
+- The transaction type is send.
+- The from address has flag 0x2
+
+Pseudocode:
+```
+func whitelistValidation(addr, tx) error {
+    if tx.Type != "send" {
+        return nil
+    }
+    if ! isSender(tx, addr) {
+        return nil
+    }
+    if ! isRecipientWhitelisted(addr, tx) {
+        return err("recipient not whitelisted")
+    }
+    return nil
+}
+```
+
+
+#### New Transaction: WhitelistRecipients
+A node should reject the transaction if any of the accounts are already whitelisted by this sender.
+
+Parameters for adding whitelisted recipients:
```

Comment:
We need to allocate extra space for whitelist storage. It will be a scability issue.